src/main/groovy/org/ysb33r/gradle/gradletest/TestGenerator.groovy

```diff
@@ -306,6 +321,90 @@ class TestGenerator extends DefaultTask {
      */
     @CompileDynamic
     private void copy(
+            final File targetDir,
+            final String testBase,
+            final String defaultTask,
+            final File manifestFile,
+            final File workDir,
+            final TestDefinition testDefinitions,
+            final List<String> arguments,
+            final boolean willFail,
+            final boolean deprecationMessageMode
+    ) {
+        copySub(testDefinitions.groovyBuildFiles,
+                GROOVY_TEST_POSTFIX,
+                targetDir,
+                testBase,
+                defaultTask,
+                manifestFile,
+                workDir,
+                testDefinitions,
+                arguments,
+                willFail,
+                deprecationMessageMode
+        )
+
+        copySub(testDefinitions.kotlinBuildFiles,
+                KOTLIN_TEST_POSTFIX,
+                targetDir,
+                testBase,
+                defaultTask,
+                manifestFile,
+                workDir,
+                testDefinitions,
+                arguments,
+                willFail,
+                deprecationMessageMode
+        )
+
+    }
+
+    private void copySub(
+            List<File> buildFiles,
+            final String postfix,
+        final File targetDir,
+        final String testBase,
+        final String defaultTask,
+        final File manifestFile,
+        final File workDir,
+        final TestDefinition testDefinitions,
+        final List<String> arguments,
+        final boolean willFail,
+        final boolean deprecationMessageMode
+    ){
+        for (File buildFile in buildFiles){
+            String testName = testBase + buildFile.name.split("\\.")[0] + postfix
+            List<String> workerArguments = arguments
+            workerArguments.addAll(["--build-file", buildFile.name])
```

Comment:
This is the magic line that makes everything else possible.  This adds the build file to the gradle command line arguments.  Without the --build-file argument, gradle looks for a build.gradle file.  this command line argument tells gradle to use a build file other than build.gradle.